src/Native/System.Security.Cryptography.Native/pal_ecc_import_export.cpp

```diff
@@ -0,0 +1,513 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <openssl/ec.h>
+#include <openssl/objects.h>
+#include "pal_ecc_import_export.h"
+#include "pal_utilities.h"
+
+ECCurveType MethodToCurveType(EC_METHOD* method)
+{
+    if (method == EC_GFp_mont_method())
+        return ECCurveType::PrimeMontgomery;
+
+    if (method == EC_GF2m_simple_method())
+        return ECCurveType::Characteristic2;
+
+    // Fallback including nist curves (optimized EC_GFp_nist_method and others)
+    return ECCurveType::PrimeShortWeierstrass;
```

Comment:
I did change this to special case each type of curve (1.0f) but that also means if a new type of nist optimized method comes along we may return unknown:

    if (method == EC_GFp_simple_method ||
        method == EC_GFp_nist_method ||
        method == EC_GFp_nistp224_method ||
        method == EC_GFp_nistp256_method ||
        method == EC_GFp_nistp521_method)
        return ECCurveType::PrimeShortWeierstrass;

    assert(false); // Unknown method
    return ECCurveType::Unspecified;